python/ray/__init__.py

```diff
@@ -9,9 +9,13 @@
 # raylet modules.
 
 if "pickle5" in sys.modules:
-    raise ImportError("Ray must be imported before pickle5 because Ray "
-                      "requires a specific version of pickle5 (which is "
-                      "packaged along with Ray).")
+    import pkg_resources
```

Comment:
Let's parse the pickle5 version number similar to https://github.com/ray-project/ray/blob/a76434ccde61bef853cb7655b22a5d67739a4913/python/ray/parameter.py#L245. While it is probably installed in most cases, it could be that `pkg_resources` is not available on the target (and we are already doing it manually for numpy and it has been working well so far). We can also use `logger.warning` for the warning.

Reply:
they don't have a `__version__` field however